WebsiteDataStore should know about network process when it is added to the process

szewai opened the pull request:
#### 616aa286262cb1322974b00b1e2652b7f2ca6927
<pre>
WebsiteDataStore should know about network process when it is added to the process
<a href="https://bugs.webkit.org/show_bug.cgi?id=270241">https://bugs.webkit.org/show_bug.cgi?id=270241</a>
<a href="https://rdar.apple.com/120988239">rdar://120988239</a>

Reviewed by Chris Dumez.

UI process sends all WebsiteDataStores to network process when creating it, but does not notify WebsiteDataStore about
the creation of network process. The result is that WebsiteDataStore does not know network process already exists and
will skip some operations, see the places where m_networkProcess is checked in WebsiteDataStore.cpp. This has caused
problems like tracking prevention flag is not updated correctly in network process.

API test: ResourceLoadStatistics.EnableResourceLoadStatisticsAfterNetworkProcessCreation

* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::sendCreationParametersToNewProcess):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::setNetworkProcess):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/275506@main">https://commits.webkit.org/275506@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/347b2bc500a693ca4465f65105810d2afe84c694

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42029 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21047 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44423 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44616 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38141 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44336 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/24225 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18375 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42603 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17975 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36193 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/15757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15647 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/37224 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46042 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38219 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37551 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/41466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16845 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13844 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/40009 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18464 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9415 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18524 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18109 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:
```diff
@@ -1670,3 +1670,64 @@ HTTPServer httpServer({
+    [websiteDataStore2 _setResourceLoadStatisticsEnabled:YES];
+    done = false;
+    [websiteDataStore2 _getResourceLoadStatisticsDataSummary:^(NSArray<_WKResourceLoadStatisticsThirdParty *> *thirdPartyData) {
+        EXPECT_EQ(static_cast<int>([thirdPartyData count]), 1);
```
Wouldn't this have worked?
```
EXPECT_EQ([thirdPartyData count], 1U);
```

szewai replied:
Yes it should work; will update it

cdumez reviewed (approved):
Still LGTM

szewai committed:
WebsiteDataStore should know about network process when it is added to the process
https://bugs.webkit.org/show_bug.cgi?id=270241
rdar://120988239

Reviewed by Chris Dumez.

UI process sends all WebsiteDataStores to network process when creating it, but does not notify WebsiteDataStore about
the creation of network process. The result is that WebsiteDataStore does not know network process already exists and
will skip some operations, see the places where m_networkProcess is checked in WebsiteDataStore.cpp. This has caused
problems like tracking prevention flag is not updated correctly in network process.

API test: ResourceLoadStatistics.EnableResourceLoadStatisticsAfterNetworkProcessCreation

* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::sendCreationParametersToNewProcess):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::setNetworkProcess):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:
(TEST):

Canonical link: https://commits.webkit.org/275506@main
